src/main/java/io/fabric8/jenkins/openshiftsync/CredentialsUtils.java

```diff
@@ -118,24 +118,38 @@ public static String upsertCredential(Secret secret) throws IOException {
 
     private static String upsertCredential(Secret secret, String namespace, String secretName) throws IOException {
         String id = null;
+        String originalId = null;
         String customSecretName = getCustomName(secret);
         if (secret != null) {
             Credentials creds = secretToCredentials(secret);
             if (creds == null)
                 return null;
+            // checking with updated secret name if custom name is not null
             id = secretName(namespace, secretName, customSecretName);
+            originalId = secretName(namespace, secretName, null);
             Credentials existingCreds = lookupCredentials(id);
+            Credentials existingOriginalCreds = lookupCredentials(originalId);
+
             final SecurityContext previousContext = ACL.impersonate(ACL.SYSTEM);
             try {
                 CredentialsStore s = CredentialsProvider.lookupStores(Jenkins.getActiveInstance()).iterator().next();
-                if (existingCreds != null) {
+                if (!originalId.equals(id)) {
+                  if (existingOriginalCreds != null){
+                    s.addCredentials(Domain.global(), creds);
+                    s.removeCredentials(Domain.global(),existingOriginalCreds);
+                    logger.info("Updated credential " + originalId + " with new Id " + id + " from Secret " + NamespaceName.create(secret)
+                      + " with revision: " + secret.getMetadata().getResourceVersion());
+                  }
+                } else {
+                  if (existingCreds != null) {
                     s.updateCredentials(Domain.global(), existingCreds, creds);
                     logger.info("Updated credential " + id + " from Secret " + NamespaceName.create(secret)
-                            + " with revision: " + secret.getMetadata().getResourceVersion());
-                } else {
+                      + " with revision: " + secret.getMetadata().getResourceVersion());
+                  } else {
                     s.addCredentials(Domain.global(), creds);
```

Comment:
I have the impression that in both cases of the if/else we do : ` s.addCredentials(Domain.global(), creds);`